chain/cosmos/client.go

```diff
@@ -155,3 +155,26 @@ func (client *Client) AccountNumber(_ context.Context, addr address.Address) (pa
 
 	return pack.U64(acc.GetAccountNumber()), nil
 }
+
+// AccountBalance returns the account balancee for a given address.
+func (client *Client) AccountBalance(_ context.Context, addr address.Address, denom string) (pack.U256, error) {
```

Comment:
We should probably have `coinDenom` as a field in `cosmos.Client`? So that we can have a uniform method signature, potentially adding it to the `Account` API.

Reply:
Yep, I imagine we would need to update the `terra.NewClient` function signature to accept the `coinDenom` if we want to add it to the API.

Also is there any reason these functions are only exposed through the `bitcoin.Client` and not apart of the overarching UTXO API?

```go
// UnspentOutputs spendable by the given address.
UnspentOutputs(ctx context.Context, minConf, maxConf int64, address address.Address) ([]utxo.Output, error)
// Confirmations of a transaction in the Bitcoin network.
Confirmations(ctx context.Context, txHash pack.Bytes) (int64, error)
```

Reply:
Tagging @loongy in case he was in charge of the UTXO API.

Reply:
> Yep, I imagine we would need to update the terra.NewClient function signature to accept the coinDenom if we want to add it to the API.

Yes.

> is there any reason these functions are only exposed

`UnspentOutputs` makes use of the [listunspent](https://bitcoin-rpc.github.io/en/doc/0.17.99/rpc/wallet/listunspent/) RPC call, which only allows fetching UTXOs for an address that is watched by our node (So it won't work for any address). Also, it scans the entire chain to fetch those UTXOs, so we should ideally avoid it, unless we need it for test purposes (which we use for our multichain test suite as well as darknode integration tests).
We could potentially add the `Confirmations` method to the UTXO API (if there is a use case). At the moment we don't need it in the backend as RenVM receives a tx only after it has sufficient number of block confirmations.